File: javascript/validParenthesis.js
const isValid = (s) => {
  const data = [];

  const checkClose = {
    ")": "(",
    "}": "{",
    "]": "[",
  };
  const strlen = s.length;

  for (var i = 0; i < strlen; ++i) {
    const str = s[i];
    if (str in checkClose == true) {
      if (data && data[data.length - 1] == checkClose[str]) {
        data.pop();
      } else {
        return false;
      }
    } else {
      data.push(str);
    }
  }
  return data.length ? false : true;
};

File: python/validParenthesis.py
def checkValidParenthesis(s: str) -> bool:
    stack = []
    checkObj = {
        ")": "(",
        "}": "{",
        "]": "["
    }

    for data in s:
       # print(data)
        if data in checkObj:
            print(data)
            if stack and stack[-1] == checkObj[data]:
                stack.pop()
            else:
                return False
        else:
            stack.append(data)
    return True if not stack else False


print(checkValidParenthesis("[()]"))
